### PR TITLE
feat(cli): enabled XCM transactions sending using the CLI

### PR DESCRIPTION
**Pull request by chexware**
### Summary by CodeRabbit

- New Feature: Introduced XCM (Cross-Chain Messaging) transactions support in the CLI. Users can now send XCM transfers using the new `xcmTransfer` command.
- New Feature: Added new required options `--signer`, `--type`, and `--endpoint` to the `xcmTransfer` command for specifying transaction details.
- Improvement: Enhanced validation for the `xcmTransfer` command parameters, ensuring better error handling and user experience.

### CODE REVIEW

**Review by MaciejBaj — APPROVED** [review]

File: client/packages/cli/src/commands/xcmTransfer/index.ts [hunk 1]
```diff
@@ -36,21 +38,76 @@ export const handleXcmTransferCommand = async (
...
-            "ROC",
-            "para"
-        )
-        console.log("\n")
-        console.table(JSON.stringify(xcmTransactionParams))
+        const xcmBeneficiaryParam = createBeneficiary(targetApi, args.recipient)
+        const xcmAssetFeeItem = targetApi.registry.createType("u32", 0)
+        console.log("Sending XCM Transfer:\n")
+        const keyring = new Keyring({ type: "sr25519" })
+        const signer = keyring.addFromUri("//Alice")
```
Comment (3h4x) [inline client/packages/cli/src/commands/xcmTransfer/index.ts:47]: keyring should be configurable, preferably from env variable. we already use CIRCUIT_SIGNER_KEY for other commands

**Review by 3h4x — APPROVED** [review]
👍

File: client/packages/cli/src/commands/xcmTransfer/index.ts (hunk 1 above)
Comment (chexware) [reply client/packages/cli/src/commands/xcmTransfer/index.ts:47]: Yup adding that shortly!

Comment (chexware) [reply client/packages/cli/src/commands/xcmTransfer/index.ts]: So added a signer value parameter that can use a custom signer value at [4e9e997](https://github.com/t3rn/t3rn/pull/1350/commits/4e9e997903a5c2b3ce1eb74053778ba0cdd25dbb) . Currently fixing the part that will allow it to use "//Circuit" as a parameter to sign XCM  transactions with CIRCUIT_SIGNER_KEY

Comment (chexware) [reply client/packages/cli/src/commands/xcmTransfer/index.ts]: all errors fixed in [dba8192](https://github.com/t3rn/t3rn/pull/1350/commits/dba8192a1292d9bc3a5b2778ac3b77bf4fabe30d)

**Pull request merged**